base_layer/core/src/transaction.rs

```diff
@@ -44,36 +58,124 @@ bitflags! {
     }
 }
 
+/// This macro implements the Ord, PartialOrd, PartialEq and Eq traits for the Hashable struct passed in
+macro_rules! hashable_ord {
+    ($hashable:ident) => {
+        impl<'a> Ord for $hashable<'a> {
+            fn cmp(&self, other: &$hashable) -> Ordering {
+                self.hash().cmp(&other.hash())
+            }
+        }
+        impl<'a> PartialOrd for $hashable<'a> {
+            fn partial_cmp(&self, other: &$hashable) -> Option<Ordering> {
+                Some(self.cmp(other))
+            }
+        }
+        impl<'a> PartialEq for $hashable<'a> {
+            fn eq(&self, other: &$hashable) -> bool {
+                self.hash() == other.hash()
+            }
+        }
+        impl<'a> Eq for $hashable<'a> {}
+    };
+}
+
+#[derive(Debug, Error)]
+pub enum TransactionError {
+    // Error validating the transaction
+    ValidationError,
+}
+
 /// A transaction input.
 ///
 /// Primarily a reference to an output being spent by the transaction.
 #[derive(Debug, Clone)]
-pub struct TransactionInput {
+pub struct TransactionInput<'a> {
     /// The features of the output being spent. We will check maturity for coinbase output.
     pub features: OutputFeatures,
     /// The commit referencing the output being spent.
-    pub commit: Commitment,
+    pub commit: Commitment<'a>,
```

Comment:
Not in this PR, but this should probably be named `commitment`

Reply:
ok